File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res;

        for(int i = 0; i < s.length(); i++){
            int len = res.size();
            if(len >= 2 && res[len - 1] == s[i] && res[len - 2] == s[i])
               continue;
            res += s[i];
        }
        return res;
    }
};

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution(object):
    def makeFancyString(self, s):
        """
        :type s: str
        :rtype: str
        """
        res = []

        for letter in s:
            if len(res) >= 2 and res[-1] == res[-2] == letter:
                continue
            res.append(letter)

        return ''.join(res)
        